metacat-metadata/src/main/java/com/netflix/metacat/metadata/store/data/entities/BaseEntity.java

```diff
@@ -0,0 +1,54 @@
+package com.netflix.metacat.metadata.store.data.entities;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.Version;
+
+/**
+ * Represents a basic metadata entity.
+ *
+ * @author rveeramacheneni
+ */
+@MappedSuperclass
+@Getter
+@Setter
```

Comment:
Would it be a better pattern to use the `Setter` tag more sparingly (apply to specific fields instead of the whole class) and prefer immutability when possible, eg for fields like `id` which shouldn't be manually set.

Reply:
That is true, allowing for setters might help with testing but fields like `id` could perhaps be always immutable. Will update.